### PR TITLE
Initial Proposal for README reorg

### PR DESCRIPTION
**Pull request by nnegrey**
@davidcavazos (Can't add you as Reviewer) 

Hi everyone, here is my proposal for the README re-org and maybe how we can target our samples going forward.

General Idea for re-org:

How to do 'X' with 'Y' technology provided by GCP

Example:
How to do Reinforcement Learning on my Image dataset with scikit-learn using ML Engine

Key things from task:
X - Reinforcement Learning
Y - ML Engine Training
Dataset type: Images
Framework: scikit-learn

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/cloudml-samples/230)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,82 @@
...
+If you don’t see something for the task you’re trying to complete, please head down to our section “What do you want to see?”
+
+# Model Training with Cloud ML Engine
+
+Notebook Tutorial:
+* [scikit-learn: Random Forest Classifier](sklearn/notebooks/ml_engine_training.ipynb) - How to train a Random Forest Classifier in scikit-learn using a text based dataset, Census, to predict a person’s income level.
+ * [XGBoost](xgboost/notebooks/XGBoost%20training%20with%20ML%20Engine.ipynb) - How to train an XGBoost model using a text based dataset, Census, to predict a person’s income level.
+
+Traditional Code Guide:
+* [Tensorflow: Linear Classifier with Stochastic Dual Coordinate Ascent (SDCA) optimizer / Deep Neural Network Classifier](criteo_tft) - How to train a Linear Classifier with SDCA and a DNN using a text based dataset, Criteo, to predict how likely a person is to click on an advertisement.
```
[the remark this answers is not in the document]
Comment (nnegrey) [reply README.md]: My idea of text based is more like not audio, video, or image based. (More high-level) to give people an idea of how they could take their data and find a sample that could get them started. 

What about: "using a text (discrete feature) based dataset"?

File: README.md [hunk 2]
```diff
@@ -1,48 +1,94 @@
...
 
 * [Genomics Ancestry Inference](https://github.com/googlegenomics/cloudml-examples) - Genomics ancestry inference using 1000 Genomes dataset
 
-- - -
+# What do you want to see?
+
+If you came looking for a sample we don’t have, please file an issue on this repository and add the label `SAMPLE_REQUEST`. Please provide as much detail as possible about the ML Engine sample you were looking for, what framework (Tensorflow, Keras, scikit-learn, XGBoost, ...), the type of model, and what kind of dataset you were hoping to use! 
+
+Sample request template:
+```
```
Comment (puneith) [inline README.md]: Make this a GH template

File: README.md [hunk 3]
```diff
@@ -1,48 +1,94 @@
...
+Framework (Tensorflow, Keras, scikit-learn, XGBoost, ...):
+Model:
+Dataset:
+Detailed information:
+```
+
+Jump below if you want to contribute and add that missing sample.
 
-## Contrib
+# How to Contribute
```
Comment (puneith) [inline README.md]: In addition to this we should have them take GH issues for new feature requests. If we have one example GH issue that will be nice and we should link the appropriate GH label (SAMPLE_REQUEST) link here.

File: README.md (hunk 2 above)
Comment (nnegrey) [reply README.md]: https://github.com/GoogleCloudPlatform/cloudml-samples/pull/232

File: README.md (hunk 3 above)
Comment (nnegrey) [reply README.md]: Added

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (elmer-garduno) [reply README.md]: how about "using a tabular (discrete feature) based dataset"? My issue with saying "text" is that it sounds more like free form text as in NLP tasks. wdyt?

**Comment by sirtorry** [comment]
nit: under model training with ML Engine, "Cloud TPUs on ML Engine" is bolded but the lines above don't fall under anything bolded. maybe label them as CPU/GPU?

same issue for "Hyperparameter Tuning (HP Tuning)"

File: CLOUD_TPU_README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Cloud TPU
+If you’re looking for samples for how to use Cloud TPU, check out the guides here. 
+
```
Comment (dizcology) [inline CLOUD_TPU_README.md]: will this file be moved under the `/tpu` directory as its root readme?

Note: we do have two tpu samples `/tpu/training` and `tpu/hptuning` now.  please add them here (or i can do it after your PR is merged)

File: README.md [hunk 5]
```diff
@@ -1,48 +1,86 @@
...
-* [Advanced Inception v3 on Cloud TPU](https://cloud.google.com/tpu/docs/tutorials/inception-v3-advanced)
-* [RetinaNet on Cloud TPU](https://cloud.google.com/tpu/docs/tutorials/retinanet)
-* [Transformer with Tensor2Tensor on Cloud TPU](https://cloud.google.com/tpu/docs/tutorials/transformer)
-
-- - -
+# Google Cloud Machine Learning Engine
+
+Welcome to the [Google Cloud Machine Learning Engine (Cloud ML Engine)](https://cloud.google.com/ml-engine/docs/) sample code repository. This repository contains samples for how to use Cloud ML Engine for model training and serving.
+
+Note: If you’re looking for our guides on how to do Machine Learning on Google Cloud Platform (GCP) without using Cloud ML Engine, please checkout our other repository: [ML on GCP](https://github.com/GoogleCloudPlatform/ml-on-gcp), which has guides on how to bring your code from various ML frameworks to [Google Cloud Platform](https://cloud.google.com/) using things like Google Compute Engine or Kubernetes.
```
Comment (dizcology) [inline README.md]: perhaps rephrase the part of "...without using Cloud ML Engine..." and say something like using other services on GCP.

File: README.md [hunk 6]
```diff
@@ -1,48 +1,86 @@
...
+
+# Getting Started
+
+If this is your first time using [Cloud ML Engine](https://cloud.google.com/ml-engine/docs/), we suggest you take a look at the [Introduction to ML Engine](https://cloud.google.com/ml-engine/docs/technical-overview) docs to get started.
+
+# Overview
+The repository is organized by tasks (Training, Serving, Training and Serving) and each task can be broken down to general usage (CPU/GPU) to specific features: Hyperparameter Tuning and TPUs. Scroll down to see what we have available, each task may provide a code or notebook solution. Where the code solution will have a README guide and the notebook solution is a full walkthrough. Our code guides are designed to provide you with the code and instructions on how to run the code, but leave you to do the digging, where our notebook tutorials try to walk you through the whole process by having the code available in the notebook throughout the guide.
+
+If you don’t see something for the task you’re trying to complete, please head down to our section “What do you want to see?”
```
Comment (dizcology) [inline README.md]: add link to that section.

File: README.md [hunk 7]
```diff
@@ -1,48 +1,86 @@
...
+* [Tensorflow: Hypertune - ResNet](tpu/hptuning/resent-hypertune) - How to run hyperparameter tuning jobs on Cloud Machine Learning Engine with Cloud TPUs using the cloudml-hypertune package.
+
+# Model Serving (Online Predictions)
+
+Notebook Tutorial:
+* [scikit-learn: Model Serving](sklearn/notebooks/Online%20Prediction%20with%20scikit-learn.ipynb) - How to train a Random Forest Classifier in scikit-learn on your local machine using a text based dataset, Census, to predict a person’s income level and deploy it on Cloud ML Engine to create predictions.
+* [XGBoost: Model Serving](xgboost/notebooks/Online%20Prediction%20with%20XGBoost.ipynb) -  How to train an XGBoost model on your local machine using a text based dataset, Census, to predict a person’s income level and deploy it on Cloud ML Engine to create predictions.
+
+Code Guide:
+
```
Comment (dizcology) [inline README.md]: this section is empty - perhaps say something like COMING SOON here.

File: README.md [hunk 8]
```diff
@@ -1,48 +1,86 @@
...
+* [Tensorflow Custom Estimator: Deep Neural Network](census/customestimator) - How to train a DNN using Tensorflow’s Custom Estimators using a text based dataset, Census, to predict a person’s income level. [TensorFlow Custom Estimator](https://www.tensorflow.org/programmers_guide/estimators#custom_estimators), which is when you write your own model function. 
+* [Tensorflow: Deep Neural Network](census/tensorflowcore) - How to train a DNN using TensorFlow’s low level APIs to create your DNN model on a single node using a text based dataset, Census, to predict a person’s income level.
+* [Tensorflow: Matrix Factorization / Deep Neural Network with Softmax](movielens) - How to train a Matrix Factorization and DNN with Softmax using a text based dataset, MovieLens 20M, to make movie recommendations.
+
+# Cloud TPU
+
+Please see the [Cloud TPU guide](CLOUD_TPU_README.md) for how to use Cloud TPU. 
+
+
+# Resources
```
Comment (dizcology) [inline README.md]: this is a little generic, and the two items under the resources section are quite different.  should the ancestry sample be under (batch?) serving?

File: README.md [hunk 9]
```diff
@@ -1,48 +1,86 @@
...
+* [Tensorflow Custom Estimator: Deep Neural Network](census/customestimator) - How to train a DNN using Tensorflow’s Custom Estimators using a text based dataset, Census, to predict a person’s income level. [TensorFlow Custom Estimator](https://www.tensorflow.org/programmers_guide/estimators#custom_estimators), which is when you write your own model function. 
+* [Tensorflow: Deep Neural Network](census/tensorflowcore) - How to train a DNN using TensorFlow’s low level APIs to create your DNN model on a single node using a text based dataset, Census, to predict a person’s income level.
+* [Tensorflow: Matrix Factorization / Deep Neural Network with Softmax](movielens) - How to train a Matrix Factorization and DNN with Softmax using a text based dataset, MovieLens 20M, to make movie recommendations.
+
+# Cloud TPU
+
+Please see the [Cloud TPU guide](CLOUD_TPU_README.md) for how to use Cloud TPU.
```
Comment (dizcology) [inline README.md]: perhaps link to `/tpu` instead.

File: README.md [hunk 10]
```diff
@@ -1,48 +1,86 @@
...
+* [Tensorflow: Matrix Factorization / Deep Neural Network with Softmax](movielens) - How to train a Matrix Factorization and DNN with Softmax using a text based dataset, MovieLens 20M, to make movie recommendations.
+
+# Cloud TPU
+
+Please see the [Cloud TPU guide](CLOUD_TPU_README.md) for how to use Cloud TPU. 
+
+
+# Resources
+
+* [TensorFlow Estimator Trainer Package Template](cloudml-template) - When training a Tensorflow model, you have to create a trainer package, here we have a template that simplifies creating a trainer package for Cloud ML Engine.
```
Comment (dizcology) [inline README.md]: maybe a template section, which could house the link to this pr after it is merged? https://github.com/GoogleCloudPlatform/cloudml-samples/pull/239

File: CLOUD_TPU_README.md (hunk 4 above)
Comment (nnegrey) [reply CLOUD_TPU_README.md]: No, since these samples aren't actually using CMLE for TPU use, but rather are the docs pages tutorials for the TPU product. The idea was to move them away from the CMLE README but still link to this README in the main page README.

Comment (nnegrey) [reply CLOUD_TPU_README.md]: The 2 TPU samples are linked in the main README.

File: README.md (hunk 5 above)
Comment (nnegrey) [reply README.md]: Updated

File: README.md (hunk 6 above)
Comment (nnegrey) [reply README.md]: Added

File: README.md (hunk 7 above)
Comment (nnegrey) [reply README.md]: Oops, Placeholder.

I'm gonna delete and if we create a guide, we can add it in.

File: README.md (hunk 10 above)
Comment (nnegrey) [reply README.md]: How about renaming Resources to Templates and make Google Samples a higher-level heading.

File: README.md [hunk 11]
```diff
@@ -1,48 +1,86 @@
...
+* Tensorflow to TF-Lite
 
 ## Google Samples
 
 * [Genomics Ancestry Inference](https://github.com/googlegenomics/cloudml-examples) - Genomics ancestry inference using 1000 Genomes dataset
```
Comment (davidcavazos) [inline README.md]: I would love to see also a section for external (unofficial) samples or docs where we link other material that could also help our users, that way if anyone discovers good material (either code or docs) we can add it here and make everyone else's life easier.

File: CLOUD_TPU_README.md (hunk 4 above)
Comment (davidcavazos) [reply CLOUD_TPU_README.md]: Can we then make this file more generic rather than being TPU-specific? This way we can add more material from different related topics

Comment (nnegrey) [reply CLOUD_TPU_README.md]: Note of offline discussion:
- Add new additional Resources section
- Leave this README for TPU
- Example: when new README for community work is added, create link in Additional Resources
- If Additional Resources sections becomes too large, create a more general README to hold this information.

File: README.md [hunk 12]
```diff
@@ -1,48 +1,82 @@
...
 
-- - -
+# Additional Resources
+
+**Cloud TPU:** Please see the [Cloud TPU guide](tpu) for how to use Cloud TPU.
+
+# What do you want to see?
+
+If you came looking for a sample we don’t have, please file an issue using the [Sample / Feature Request](https://github.com/GoogleCloudPlatform/cloudml-samples/issues/new?template=sample-feature-request.md) template on this repository. Please provide as much detail as possible about the ML Engine sample you were looking for, what framework (Tensorflow, Keras, scikit-learn, XGBoost, ...), the type of model, and what kind of dataset you were hoping to use!
```
Comment (davidcavazos) [inline README.md]: This is awesome, thanks!

File: README.md [hunk 13]
```diff
@@ -1,48 +1,82 @@
...
-* [Transformer with Tensor2Tensor on Cloud TPU](https://cloud.google.com/tpu/docs/tutorials/transformer)
-
-- - -
-
-## Google Samples
+# Google Cloud Machine Learning Engine
+
+Welcome to the [Google Cloud Machine Learning Engine (Cloud ML Engine)](https://cloud.google.com/ml-engine/docs/) sample code repository. This repository contains samples for how to use Cloud ML Engine for model training and serving.
+
+Note: If you’re looking for our guides on how to do Machine Learning on Google Cloud Platform (GCP) using other services, please checkout our other repository: [ML on GCP](https://github.com/GoogleCloudPlatform/ml-on-gcp), which has guides on how to bring your code from various ML frameworks to [Google Cloud Platform](https://cloud.google.com/) using things like Google Compute Engine or Kubernetes.
```
Comment (davidcavazos) [inline README.md]: Add links to `Google Compute Engine` and `Kubernetes`
